server/routers/api/decisionlog.ts

```diff
@@ -0,0 +1,129 @@
+
+const apicache = require('apicache');
+const mongoose = require('mongoose');
+import { logger, getNewJudgementCounts, asyncHandler } from '../../common';
+
+export const decisionLogRouter = require('express').Router();
+
+type DecisionLogHookFunc = (i: DecisionLogProps) => any;
+
+const hooks = {};
+export const installHook = function(hookName, hookFunc:DecisionLogHookFunc) {
+  hooks[hookName] = hookFunc;
+  logger.info(`Hook ${hookName} is installed`);
+}
+
+async function getPreviousWarningsAuthor(
+  userId: string,
+  endTimestamp: number,
+  warningTimeframe: number,
+  warningThreshold: number,
+) {
+  var endDate = new Date(endTimestamp);
+  let startTimestamp = new Date();
+  startTimestamp.setDate(endDate.getDate() - warningTimeframe);
+  let eventsBeforeEnd: DecisionLogProps[] = await DecisionLog.find({
+    userId: userId,
+    type: {$in: ["warning", "block"]},
+    timestamp: {$gte: startTimestamp, $lte: endDate}
+  }).sort([['timestamp', 1]]).limit(warningThreshold * 10); //Avoid too many events 
+  console.log("Get " + eventsBeforeEnd.length + " past events for " + userId);
+  return eventsBeforeEnd;
+}
+
+async function getPreviousWarningsArticle(
+  title: string, 
+  endTimestamp: number,
+  warningTimeframe: number,
+  warningThreshold: number,
+) {
+  var endDate = new Date(endTimestamp);
+  let startTimestamp = new Date();
+  startTimestamp.setDate(endDate.getDate() - warningTimeframe);
+  let eventsBeforeEnd: DecisionLogProps[] = await DecisionLog.find({
+    title: title,
+    type: {$in: ["protect", "articleLogEvent"]},
+    timestamp: {$gte: startTimestamp, $lte: endDate}
+  }).sort([['timestamp', 1]]).limit(warningThreshold * 10); //Avoid too many events 
+  console.log("Get " + eventsBeforeEnd.length + " past events for " + title);
+  return eventsBeforeEnd;
+}
+
+const getAuthorLog = async (req, res) => {
+  let userId = req.params.userId;
+  let endTimestamp = req.params.endTimestamp;
+  let warningTimeframe = req.params.warningTimeframe;
+  let warningThreshold = req.params.warningThreshold;
+  var events = await getPreviousWarningsAuthor(userId, endTimestamp, warningTimeframe, warningThreshold);
+  res.send(events);
+  req.visitor
+      .event({ ec: "api", ea: "/decisionLog/author/:userId" })
+      .send();
+};
+decisionLogRouter.get(`/author/:userId/:endTimestamp/:warningTimeframe/:warningThreshold`, asyncHandler(getAuthorLog));
+
+const getArticleLog = async (req, res) => {
+  let title = req.params.title;
+  let endTimestamp = req.params.endTimestamp;
+  let warningTimeframe = req.params.warningTimeframe;
+  let warningThreshold = req.params.warningThreshold;
+  var events = await getPreviousWarningsArticle(title, endTimestamp, warningTimeframe, warningThreshold);
+  res.send(events);
+  req.visitor
+      .event({ ec: "api", ea: "/decisionLog/article/:title" })
+      .send();
+};
+decisionLogRouter.get(`/article/:title/:endTimestamp/:warningTimeframe/:warningThreshold`, asyncHandler(getArticleLog));
```

Comment:
Can you use url `query` instead? Having such a long parameter define in the `path` (`/article/:title/:endTimestamp/:warningTimeframe/:warningThreshold`) has a lot of disadvantages, such as it's ordered, and all information needs to be in place, as opposed to url query which allows much more flexibility.